### ISSUE TITLE
GPS map not working

### ISSUE THREAD

**Issue by Massimo-B** [BestImageViewer/geeqie#996]
### ISSUE TYPE

* Bug Report

### GEEQIE VERSION

```
1.7.3
```

```

```

### OS / DISTRIBUTION

Gentoo

### SUMMARY
GPS map is not working

### STEPS TO REPRODUCE

Starting geeqie, opening a picture with GPS data.
Using the + button to add the panel "GPS Map".
Nothing happens.

**Comment by caclark** [comment]
Is the problem as before - no map data is shown? If you right-click on the map pane are there 9 map options displayed?

**Comment by Massimo-B** [comment]
No, I don't see the map pane at all. I tried scaling and moving but it does not seem to exist.

**Comment by caclark** [comment]
Please try this:
Open `Edit/Preferences/Behavior` and select `Override disable GPU`
Exit `Preferences`
Exit Geeqie
Restart Geeqie.

Also, note that since recently the default map of Mapnik is not available. It is necessary to use Cycle Map or Transport Map (via right-click).
That is another problem.

**Comment by yarda** [comment]
> Please try this: Open `Edit/Preferences/Behavior` and select `Override disable GPU` Exit `Preferences` Exit Geeqie Restart Geeqie.
> 
> Also, note that since recently the default map of Mapnik is not available. It is necessary to use Cycle Map or Transport Map (via right-click). That is another problem.

I had the same problem, this workaround worked for me, Fedora 35, geeqie-1.7.2, libchamplain-0.12.20.

**Comment by caclark** [comment]
> This has solved the issue, thanks. What disadvantage do I have with disabled GPU?

It is the other way around. The use of the Clutter library was being disabled on your system, and that disabled display of the GPS maps. 
The option you have selected overrides the disable. There is now no need for your concern.

[The origin of the problem was discussed in https://github.com/BestImageViewer/geeqie/issues/397 - it is a bit of a mess that needs to be sorted out - sometime]

**Comment by tomaszg7** [comment]
I also had a problem getting GPS map working a few days ago even though the `Override` was enabled. Not sure what I did but now it works ;) Although there is some minor bug - adding the map widget to the sidebar (even if map is already present) resizes the sidebar to something around 70% of window width.

**Comment by caclark** [comment]
> Although there is some minor bug - adding the map widget to the sidebar (even if map is already present) resizes the sidebar to something around 70% of window width.

#999

**Comment by gusnan** [comment]
I've got a bug on the Debian package that geeqie shouldn't depend on the clutter library - It's getting removed from Debian:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1018123

(The Debian package depends on libchamplain, which depends on clutter).

I have removed the depends from the debian package, but it would of course be better to have a proper replacement.

Upstream bug here: https://gitlab.gnome.org/GNOME/Initiatives/-/issues/31

It looks like for GTK4 you'll have libshumate (https://gitlab.gnome.org/GNOME/libshumate) as a replacement for libchamplain.

**Comment by caclark** [comment]
I am unclear what the process is for this.
The file `meson_options.txt` has "auto" as the default for "gps_map". If the user has the clutter library installed, it will be used, otherwise it will be ignored.
If the user does not want to use clutter, the option should be set to "disabled".
Is there something else you would like done?

[GTK4 for Geeqie is just a dream - or maybe a nightmare]

**Comment by gusnan** [comment]
@caclark Nah, as you see it seems like there unfortunately isn't that much can be done at the moment - I posted mostly here to keep you guys informed.

And I realize that a port to GTK4 is quite some time away still (and I don't have anything against that).

**Comment by gdt** [comment]
I was also having the missing map.   I don't really understand "override disable GPU".   I would guess there is normal code and it works and there is code to use the GPU and it fails, but using the GPU is defaulting to on.   it seems we should default to off.
